.github/workflows/build-and-publish.yml

```diff
@@ -84,5 +88,56 @@ jobs:
         with:
           upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
           asset_path: ./app/bundled_app/linux/${{ github.sha }}
-          asset_name: pilotcli
+          asset_name: pilotcli_linux
+          asset_content_type: application/octet-stream
+
+  push-binary-macos:
+    needs: [ push-binary-linux ]
+    if: ${{ needs.extract-branch-name.outputs.branch == 'main' || needs.extract-branch-name.outputs.branch == 'develop'}}
+    runs-on: macos-12
```

Comment:
is this x86 or arm64?

Reply:
it seems like x64 version. Any concerns on this?
https://github.blog/2023-03-01-github-actions-introducing-faster-github-hosted-x64-macos-runners/
java version is also x64: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#java